### PR TITLE
Add edit page for personal schedules

### LINKED ISSUE #7: FEATURE: Add edit page for personal schedules
### Have you updated the title?

- [X] I have updated the issue's title.

### As a...

user

### I can...

click "Edit" in the personal schedule table

### So that...

I can edit my personal schedule

### Discussion

Currently the personal schedule edit page is a placeholder, replace this with a form or table to allow students to delete courses from their personal schedule.

### Acceptance Criteria

- [ ] There is a form or table which displays a specific personal schedule & allows the user to delete courses.
- [ ] This form or table has been added to the personal schedule edit page.
- [ ] Deleting courses from a personal schedule deletes them from the PSCourse repository.

### Implementation Todos

Frontend:
- [ ] Add a Personal Schedule Edit Table (or similar), which allows the user to view the courses in a schedule & delete any courses.
- [ ] This edit table should make API calls to the PSCourse API to delete courses 
- [ ] Add this edit table to the Personal Schedule Edit Page

Testing:
- [ ] Front end tests pass and there is adequate coverage
- [ ] Back end tests pass and there is adequate coverage

### PR DESCRIPTION
**Pull request by ChrisWoolson**
Added editPage for personal schedules, and its tests.
Closes #7 

Created Edit Schedules page, Edit Schedules table, Edit Schedules Utils, and their corresponding tests.
Also added editCourseFixtures for testing.

Screenshot 2024-03-07 at 12 56 27 AM

Dokku Dev Deployment: https://courses-chriswoolson.dokku-05.cs.ucsb.edu

### CODE REVIEW

**Review by winzuo — DISMISSED** [review]
LGTM

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -25,7 +25,7 @@ spring.mvc.pathmatch.matching-strategy = ANT_PATH_MATCHER
 server.compression.enabled=false
 
 spring.data.mongodb.uri=${MONGODB_URI:${env.MONGODB_URI:mongodb+srv://fakeUsername:fakePassword@cluster0.ulqcw.mongodb.net/fakeDatabase?retryWrites=true&w=majority}}
-
+spring.data.mongodb.database=database
```
Comment (tallyhawley) [inline src/main/resources/application.properties]: This is already defined at the bottom of this file. Please remove one or the other.

**Review by tallyhawley — CHANGES_REQUESTED** [review]
Just one small change, other than that looks good!

File: src/main/resources/application.properties (hunk 1 above)
Comment (ChrisWoolson) [reply src/main/resources/application.properties]: Oops thanks

**Review by JieTiao — APPROVED** [review]
LGTM

**Review by tallyhawley — APPROVED** [review]
LGTM

**Pull request merged**